tools/common/StaticRegistrar.cs

```diff
@@ -2228,15 +2230,32 @@ void Specialize (AutoIndentStringBuilder sb)
 				if (!string.IsNullOrEmpty (single_assembly) && single_assembly != @class.Type.Module.Assembly.Name.Name)
 					continue;
 
-#if !MONOMAC
 				var isPlatformType = IsPlatformType (@class.Type);
+#if !MONOMAC
 
 				if (isPlatformType && IsSimulatorOrDesktop && IsMetalType (@class))
 					continue; // Metal isn't supported in the simulator.
 
 				if (IsSimulatorOrDesktop && Driver.App.Platform == Xamarin.Utils.ApplePlatform.TVOS && IsExternalAccessoryType (@class))
 					continue; // ExternalAccessory's headers aren't available for tvOS/Simulator.
+#else
+				// Don't register 64-bit only API on 32-bit XM
+				if (!Is64Bits)
+				{
+					var attributes = GetAvailabilityAttributes (@class.Type); // Can return null list
+					if (attributes != null && attributes.Any (x => x.Architecture == PlatformArchitecture.Arch64))
+						continue;
+				}
+
+				if (IsQTKitType (@class) && GetSDKVersion () >= new Version (10,12))
+					continue; // QTKit header was removed in 10.12 SDK
+
+				// These are 64-bit frameworks that extend NSExtensionContext / NSUserActivity, which you can't do
+				// if the header doesn't declare them. So hack it away, since they are useless in 64-bit anyway
+				if (!Is64Bits && (IsMapKitType (@class) || IsIntentsType (@class)))
```

Comment:
Yes. I tried removing it. There must be another chunk of code that is handling the category code or something.

Even marking the category bits as onlyOn64 didn't fix it.

I did not look into it more as I got distracted with the extensions stuff. I can do more digging today if you desire.